app/src/main/java/com/brandonnalls/mockmocklocations/OverrideSettingsSecure.java

```diff
@@ -10,14 +11,27 @@
 
 public class OverrideSettingsSecure implements IXposedHookLoadPackage {
     public void handleLoadPackage(final LoadPackageParam lpparam) throws Throwable {
+        // before API level 18, hook Settings.Secure.ALLOW_MOCK_LOCATION
+        if (Build.VERSION.SDK_INT < 18) {
+            findAndHookMethod("android.provider.Settings.Secure", lpparam.classLoader, "getString",
+                    ContentResolver.class, String.class, new XC_MethodHook() {
+                @Override
+                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
+                    String requested = (String) param.args[1];
+                    if (requested.equals(Settings.Secure.ALLOW_MOCK_LOCATION)) {
+                        param.setResult("0");
+                    }
+                }
+            });
+            return;
+        }
 
-        findAndHookMethod("android.provider.Settings.Secure", lpparam.classLoader, "getString", ContentResolver.class, String.class, new XC_MethodHook() {
+        // after API level 18, hook Location.isFromMockProvider
```

Comment:
The comment says after API 18, but this is getting applied to all version numbers.